vars/abapEnvironmentPipelineStageCloneRepositories.groovy

```diff
@@ -8,20 +8,25 @@ import static com.sap.piper.Prerequisites.checkScript
 @Field Set GENERAL_CONFIG_KEYS = []
 @Field STAGE_STEP_KEYS = [
     /** Pulls Software Components / Git repositories into the ABAP Environment instance */
-    'abapEnvironmentPullGitRepo'
+    'abapEnvironmentPullGitRepo',
+    /** Checks out a Branch in the pulled Software Component on the ABAP Environment instance */
+    'abapEnvironmentCheckoutBranch'
 ]
 @Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS.plus(STAGE_STEP_KEYS)
 @Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
 /**
- * This stage clones Git repositories / software components to the ABAP Environment instance
+ * This stage clones Git repositories / software components to the ABAP Environment instance and checks out the master branch
  */
 void call(Map parameters = [:]) {
     def script = checkScript(this, parameters) ?: this
 
     def stageName = parameters.stageName?:env.STAGE_NAME
 
     piperStageWrapper (script: script, stageName: stageName, stashContent: [], stageLocking: false) {
-        abapEnvironmentPullGitRepo script: parameters.script
+        if!(parameters.script.commonPipelineEnvironment.configuration.runStage?.get("Prepare System")) {
+            abapEnvironmentPullGitRepo script: parameters.script
```

Comment:
This may be a bit counter intuitive, but IMO PullGitRepo and CheckoutBranch should be changed.

In the near future, PullGitRepo should be able to checkout a specific commit in the history of the branch.

Assuming the repo is initially on "branch1" on any commit.
First, checkout will be executed. If the target is "branch1" nothing happens (as far as I know).
If the target is "branch2", the branch will be switched and the checked out commit is HEAD.
Second, "pullGitRepo" will checkout HEAD or the specified commit on the active branch.

In the end, this depends how the possibility to checkout a specific commit is implemented. But I think the outlined solution is the most likely one.

Reply:
Ok, counter intuitive but changed it if it makes future changes easier.